src/pipecat/processors/frame_processor.py

```diff
@@ -669,19 +716,24 @@ async def __resume(self, frame: FrameProcessorResumeFrame | FrameProcessorResume
     async def _start_interruption(self):
         """Start handling an interruption by cancelling current tasks."""
         try:
-            # Cancel the process task. This will stop processing queued frames.
-            await self.__cancel_process_task()
+            # Cancel the process task. This will stop processing queued
+            # frames. Note that if we were waiting for an interruption we can't
+            # cancel the process task because it's the task that is still
+            # waiting and might still need to do things.
+            if not self._wait_for_interruption:
+                await self.__cancel_process_task()
         except Exception as e:
             logger.exception(f"Uncaught exception in {self} when handling _start_interruption: {e}")
             await self.push_error(ErrorFrame(str(e)))
 
-        # Create a new process queue and task.
-        self.__create_process_task()
-
-    async def _stop_interruption(self):
-        """Stop handling an interruption."""
-        # Nothing to do right now.
-        pass
+        if not self._wait_for_interruption:
+            # Create a new process queue and task.
+            self.__create_process_task()
+        else:
```

Comment:
I feel like this `else` is conceptually more related to the `if` on line 723:

"if waiting for interruption, cancel. otherwise, reset"

Then you wouldn't need to have the slightly awkward "see comment above".

Reply:
This is what I did:

```
            if self._wait_for_interruption:
                # If we get here we know the process task was just waiting for
                # an interruption (push_interruption_task_frame_and_wait()), so
                # we can't cancel the task because it might still need to do
                # more things (e.g. pushing a frame after the
                # interruption). Instead we just drain the queue because this is
                # an interruption.
                self.__reset_process_task()
            else:
                # Cancel and re-create the process task including the queue.
                await self.__cancel_process_task()
                self.__create_process_task()
```

I left the `reset_process_task()` as is without arguments.